### PR TITLE
Small fixes: embeddings and auto-context

### PR DESCRIPTION
**Pull request by granawkins**
Required to run the benchmark_runner successfully

### CODE REVIEW

**Comment by granawkins** [comment]
I added a few other things I used to get the sampler kicking over the weekend.

File: mentat/embeddings.py [hunk 1]
```diff
@@ -135,7 +135,10 @@ async def get_feature_similarity_scores(
                 return [0.0 for _ in checksums]
 
     # Fetch embeddings in batches
-    n_batches = len(embed_texts) // EMBEDDINGS_API_BATCH_SIZE + 1
+    if len(embed_texts) == 0:
```
Comment (jakethekoenig) [inline mentat/embeddings.py:138]: Why did you make this change? It should have no logical effect right?

File: scripts/evaluate_samples.py [hunk 2]
```diff
@@ -33,6 +35,9 @@ def warn(msg: Any):
 
 
 SAMPLES_DIR = mentat_dir_path / "samples"
+os.makedirs(SAMPLES_DIR, exist_ok=True)
```
Comment (jakethekoenig) [inline scripts/evaluate_samples.py:38]: path objects have a `mkdir` which may be better to use here for consistency with the rest of the codebase.

**Review by jakethekoenig — APPROVED** [review]

**Comment by jakethekoenig** [comment]
I'm going to go ahead and merge this because I want the bug fixes and the rest of the PR isn't user facing.

**Pull request merged**